File: src/Nucleo.Web/Web/NavigationControls/NavigationBarView.cs
using System;
using System.Collections.Generic;
using System.Web.UI;
using System.Web.UI.WebControls;


namespace Nucleo.Web.NavigationControls
{
	/// <summary>
	/// Represents the view of the navigation bar selections; this has to be programmatically controlled by you at the moment.
	/// </summary>
	[WebScriptMetadata(typeof(NavigationBarViewScriptMetadata))]
	public class NavigationBarView : BaseAjaxControl
	{
		#region " Properties "

		/// <summary>
		/// Gets the ID of the container control it resides in.
		/// </summary>
		public string ContainerControlID
		{
			get { return (string)ViewState["ContainerControlID"]; }
			set { ViewState["ContainerControlID"] = value; }
		}

		#endregion



		#region " Methods "

		protected override void GetAjaxScriptDescriptors(IContentRegistrar registrar)
		{
			base.GetAjaxScriptDescriptors(registrar);

			NucleoScriptControlDescriptor descriptor = registrar.AddDescriptor<NucleoScriptControlDescriptor>(
				new ScriptDescriptionRequestDetails(this, typeof(NavigationBarView), this.ClientID));
			descriptor.AddComponentPropertyIfExists("containerControl", ControlUtility.FindControl(this, this.ContainerControlID).ClientID);
		}

		#endregion
	}
}

File: src/Nucleo.Web/Web/NavigationControls/NavigationBarView.js
Type.registerNamespace("Nucleo.Web.NavigationControls");Nucleo.Web.NavigationControls.NavigationBarView=function(a){Nucleo.Web.NavigationControls.NavigationBarView.initializeBase(this,[a]);this._containerControl=null};Nucleo.Web.NavigationControls.NavigationBarView.prototype={get_containerControl:function(){return this._containerControl},set_containerControl:function(a){this._containerControl=a;if(a!=null){a._view=this}},initialize:function(){Nucleo.Web.NavigationControls.NavigationBarView.callBaseMethod(this,"initialize")},dispose:function(){Nucleo.Web.NavigationControls.NavigationBarView.callBaseMethod(this,"dispose")}};Nucleo.Web.NavigationControls.NavigationBarView.registerClass("Nucleo.Web.NavigationControls.NavigationBarView",Nucleo.Web.BaseAjaxControl);if(typeof(Sys)!=="undefined"){Sys.Application.notifyScriptLoaded()};